src/features/transfer/utils.ts

```diff
@@ -0,0 +1,57 @@
+import { providers } from 'ethers';
+import { toast } from 'react-toastify';
+
+import { HyperlaneCore } from '@hyperlane-xyz/sdk';
+import { ProtocolType, convertDecimals } from '@hyperlane-xyz/utils';
+
+import { logger } from '../../utils/logger';
+import { getProtocolType } from '../caip/chains';
+import { isNonFungibleToken } from '../caip/tokens';
+import { AdapterFactory } from '../tokens/AdapterFactory';
+import { Route } from '../tokens/routes/types';
+import { isHypRoute, isRouteToCollateral } from '../tokens/routes/utils';
+
+// In certain cases, like when a synthetic token has >1 collateral tokens
+// it's possible that the collateral contract balance is insufficient to
+// cover the remote transfer. This ensures the balance is sufficient or throws.
+export async function ensureSufficientCollateral(route: Route, weiAmount: string) {
```

Comment:
Contents of this file aren't new. Moved here from `useTokenTransfer.ts` b.c. it's too big